### PR TITLE
Gea 1427 sign criteria

### PR DESCRIPTION
**Pull request by pangea-andrest**
- Unify sign criteria with Go SDK
- Add signature test using all fields on Event. In this case, compare signature and encoded public key manually with Go SDK.

### CODE REVIEW

File: pangea/services/audit.py [hunk 1]
```diff
@@ -617,8 +621,9 @@ def verify_signature(self, audit_envelope: JSONObject) -> bool:
         """
         sign_envelope = self.create_signed_envelope(audit_envelope.envelope.event)
         public_key_b64 = audit_envelope.envelope.public_key
-        public_key_bytes = b64decode(public_key_b64)
-        return self.sign.verifyMessage(audit_envelope.envelope.signature, sign_envelope, public_key_bytes)
+        public_key_bytes = b64decode_ascii(public_key_b64)
```
Comment (dbenders) [inline pangea/services/audit.py:624]: what's the difference between `b64decode` and `b64decode_ascii`? Why only here?

File: pangea/signing.py [hunk 2]
```diff
@@ -2,188 +2,120 @@
...
-                )
-
-                with open(self._public_key_filename, "wb") as file:
-                    file.write(public_bytes)
-
-            except Exception:
-                raise Exception("Error: Failed generating public key.")
+class Signer:
+    def __init__(self, private_key_file: str) -> None:
+        self.__private_key = None
```
Comment (dbenders) [inline pangea/signing.py:15]: two underscores are for internal use only, please use one

File: pangea/signing.py [hunk 3]
```diff
@@ -2,188 +2,120 @@
...
+        self.__private_key_filename = private_key_file
 
     # Returns the private key
-    def getPrivateKey(self, private_bytes: bytes = None):
-        if self._private_key is None and private_bytes is None:
+    def __getPrivateKey(self) -> ed25519.Ed25519PrivateKey:
```
Comment (dbenders) [inline pangea/signing.py:19]: no two underscores please

File: pangea/signing.py [hunk 4]
```diff
@@ -2,188 +2,120 @@
...
+        if self.__private_key is None:
             try:
-                self.generateKeys(self._overwrite_keys_if_exists)
-                with open(self._private_key_filename, "rb") as file:
-                    private_bytes = file.read()
+                with open(self.__private_key_filename, "rb") as file:
+                    file_bytes = file.read()
             except Exception:
-                raise Exception("Error: Failed loading private key.")
+                print(os.getcwdb())
```
Comment (dbenders) [inline pangea/signing.py:25]: remove print

File: pangea/signing.py [hunk 5]
```diff
@@ -2,188 +2,120 @@
...
+            .public_bytes(encoding=serialization.Encoding.Raw, format=serialization.PublicFormat.Raw)
+        )
 
-    # Verify a string message using Ed25519 algorithm
-    def verifyMessageStr(self, signature_b64: bytes, message: str, public_key_bytes: bytes = None) -> bool:
-        message_bytes = bytes(message, "utf8")
-        return self.verifyMessageBytes(signature_b64, message_bytes, public_key_bytes)         
+
+class Verifier:
+    def __init__(self):
```
Comment (dbenders) [inline pangea/signing.py:83]: I think there's no need to add an empty init

File: pangea/signing.py [hunk 6]
```diff
@@ -2,188 +2,120 @@
...
+        return
+
+    # verify message with signature and public key bytes
+    def verifyMessage(self, signature_b64: bytes, message: t.Any, public_key_bytes: bytes = None) -> bool:
```
Comment (dbenders) [inline pangea/signing.py:87]: ```message: str|dict|bytes```

File: pangea/services/audit.py (hunk 1 above)
Comment (pangea-andrest) [reply pangea/services/audit.py:624]: If it's not enc/dec as ascii json dumps fails

**Review by dbenders — APPROVED** [review]

**Pull request merged**